### PR TITLE
Accept DOM nodes as the tooltip / popover content

### PR DESCRIPTION
**Pull request by glebm**
When a DOM node is passed to an HTML tooltip, the `title` node is only moved if it is not already in the tooltip. Otherwise, `empty()` is used instead of `detach()` before appending the `title` to avoid memory
leaks. If a DOM node is passed to a plain text tooltip, its text is copied via jQuery `.text()`.

### CODE REVIEW

**Comment by cvrebert** [comment]
@fat Ping.

**Comment by cvrebert** [comment]
I think this needs some new unit tests to exercise the "passed a jQuery object" and "passed a DOM node" cases, but LGTM otherwise.
@XhmikosR LGTY?

**Comment by glebm** [comment]
I'lll add tests if LGTY'all.

**Comment by twbs-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 06df28fa12232cc28603c9e0d3dc957ceeb2e608
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/52668697

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by cvrebert** [comment]
This also needs to update the docs to reflect the new types that `content` and `title` are allowed to be.

**Comment by twbs-savage** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: ba4b6cc94e987da5e328d58146d62e1907427dc9
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/53870706

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by glebm** [comment]
I have moved the `setElementContent` to tooltip, and added tests.

**Comment by twbs-savage** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: c4c8a6af5244a0e2e80ff8bb80f66e3fec5d38ad
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/53871766

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by cvrebert** [comment]
IE8 :japanese_ogre:

**Comment by glebm** [comment]
⚔:japanese_ogre:

**Comment by twbs-savage** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 245bf6cb5f0a4c3b1f8a36882a3ee687df6cbaea
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/53874885

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by twbs-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 539f941c2e79df0f553bb21e325d7232f07939db
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/53874916

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

File: js/popover.js [hunk 1]
```diff
@@ -41,14 +41,10 @@
   }
 
   Popover.prototype.setContent = function () {
-    var $tip    = this.tip()
-    var title   = this.getTitle()
-    var content = this.getContent()
-
-    $tip.find('.popover-title')[this.options.html ? 'html' : 'text'](title)
-    $tip.find('.popover-content').children().detach().end()[ // we use append for html objects to maintain js events
```
Comment (cvrebert) [inline js/popover.js]: @glebm Don't we need this emptying logic?

Comment (glebm) [inline js/popover.js]: @cvrebert Right, I've added `.empty()` before append in the case where an element is being appended. I don't think we should `.detach()` as it is almost never useful but always creates memory leaks when it's not. The difference between `empty` and `detach` is that the latter keeps all the attached jQuery events and data, but since we do not return the references to previous children the user would have to keep these references themselves, thus they can `detach` themselves anyway if they need to.

**Comment by twbs-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: ba1f5488ff68372eabae599f5856f952607a7a27
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/61608267

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by glebm** [comment]
For some reason continuous-integration/travis-ci/push is associated with the wrong branch. I pushed to twbs/fix-popover-setContent instead of glebm/fix-popover-setContent by accident and then promptly deleted the twbs branch. The badge above is trying to build a non-existing branch.

**Comment by twbs-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 14ad1fa810664fa2900c049c4d75fe5686930848
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/65015394

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by twbs-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 73065a96c4251a8f3d8e6abe905fa401f78c33bd
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/65016339

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by twbs-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 8680a68a49d02b3451c47ac19382f025d1465103
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/65017392

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by glebm** [comment]
Squashed, rebased, and updated the docs to reflect the new accepted types.

**Comment by twbs-savage** [comment]
**Tests passed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **CONFIRMED**

Commit: 3857d18ee1c3dff7633eaf1de4948337bd27c1b8
Build details: https://travis-ci.org/twbs-savage/bootstrap/builds/65017979

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by cvrebert** [comment]
Perhaps we should close this since it's arguably a new feature and it's targeted at v3.

**Comment by glebm** [comment]
Alright. :+1:
